rollout/mocks/TrafficRoutingReconciler.go

```diff
@@ -14,7 +14,7 @@ type TrafficRoutingReconciler struct {
 }
 
 // SetWeight provides a mock function with given fields: desiredWeight, additionalDestinations
-func (_m *TrafficRoutingReconciler) SetWeight(desiredWeight int32, additionalDestinations ...v1alpha1.WeightDestination) error {
+func (_m *TrafficRoutingReconciler) SetWeight(desiredWeight int32, headerWeight *v1alpha1.SetHeaderWeight, additionalDestinations ...v1alpha1.WeightDestination) error {
```

Comment:
Just confirming, is this adding the ability to specify the header weight in the existing `setWeight` step? Can `WeightDestination` be used instead of adding an additional param in the signature?

Reply:
@leoluz Yes. It's adding the new route with a header based traffic.
BTW there is a mistake here: should be `SetHeaderRoute`.

We can not use the **WeightDestination** as **SetHeaderRoute** has different fields